pkg/controller/controlplanecerts/controlplanecerts_controller_test.go

```diff
@@ -203,8 +174,44 @@ func TestReconcileControlPlaneCerts(t *testing.T) {
 
 			assert.Nil(t, err)
 
-			if test.validate != nil {
-				test.validate(t, fakeClient, applier.appliedObjects)
+			cd := getFakeClusterDeployment(t, fakeClient)
+
+			if test.expectNoSyncSet {
+				assert.Len(t, applier.appliedObjects, 0, "unexpected syncset apply")
+			} else {
+				assert.Len(t, applier.appliedObjects, 1, "single apply expected")
+				assert.IsType(t, &hivev1.SyncSet{}, applier.appliedObjects[0], "syncset apply expected")
```

Comment:
should this be a `require.IsType` so that we don't bother with the next tests if this is not true?

Reply:
Made this and the line above requires.